### DISCUSSION TITLE
How can I login without redirect?

### DISCUSSION THREAD

**Discussion by moecasts** [ory/hydra#3634] [Q&A] [answered]
I want to implement the login function like kratos does, without needing a login_challenge, because if using the conventional oauth2 flow, redirection is required, which makes it impossible to log in within a popup component. Is there any way this can be achieved?

**Comment by vinckr** [comment, marked as answer]
Hello @moecasts 
Can you tell me a bit more about the popup you are trying to use ?

Generally I would discourage that, as browser venders are clamping down hard on [third party cookies](https://developer.chrome.com/en/docs/privacy-sandbox/third-party-cookie-phase-out/) and in turn also on using iframes etc. for things like this. 

So in short:

It can probably be achieved with some fiddling - but I would recommend to not do it, because it will probably break/be impossible soon.
